tools/plugins/com.liferay.ide.portlet.core/src/com/liferay/ide/portlet/core/spring/NewSpringPortletClassDataModelProvider.java

```diff
@@ -0,0 +1,186 @@
+            return false;
+        }
+        else if( CREATE_JSPS_FOLDER.equals( propertyName ) )
+        {
+            return false;
+        }
+        else if( EDIT_MODE.equals( propertyName ) )
+        {
+            return false;
+        }
+        else if( HELP_MODE.equals( propertyName ) )
+        {
+            return false;
+        }
+
+        return super.isPropertyEnabled( propertyName );
+    }
+
+    @Override
+    public Object getDefaultProperty( String propertyName )
+    {
+        if( CLASS_NAME.equals( propertyName ) )
+        {
+            return "NewSpringMVCPortlet"; //$NON-NLS-1$
+        }
+        else if( DEFAULT_VIEW_CONTROLLER_CLASS.equals( propertyName ) )
+        {
+            return DEFAULT_PORTLET_CONTROLLER_NAME;
+        }
+        else if( CREATE_JSPS_FOLDER.equals( propertyName ) )
+        {
+            return "/WEB-INF/jsp/" + getProperty( PORTLET_NAME ).toString().toLowerCase(); //$NON-NLS-1$
+        }
+        else if( SHOW_NEW_CLASS_OPTION.equals( propertyName ) )
+        {
+            return false;
+        }
+        else if( INIT_PARAMS.equals( propertyName ) )
+        {
+            return getInitParams();
+        }
+
+        return super.getDefaultProperty( propertyName );
+    }
+
+    @Override
+    protected Object getInitParams()
+    {
+        List<ParamValue> initParams = new ArrayList<ParamValue>();
+
+        String value = "/WEB-INF/spring-context/portlet/";
+
+        final String portletName = getStringProperty( PORTLET_NAME );
+
+        value = value + portletName + "-portlet.xml";
+
+        final ParamValue paramValue = CommonFactory.eINSTANCE.createParamValue();
+
+        paramValue.setName( "contextConfigLocation" );
+
+        paramValue.setValue( value );
+
+        Collections.addAll( initParams, paramValue );
+
+        return initParams;
+    }
+
+    @Override
+    public Set getPropertyNames()
+    {
+        Set propertyNames = super.getPropertyNames();
+
+        propertyNames.add( DEFAULT_VIEW_CONTROLLER_CLASS );
+
+        return propertyNames;
+    }
+
+    @Override
+    public IStatus validate( String propertyName )
+    {
+        if( PORTLET_NAME.equals( propertyName ) )
+        {
+            IStatus status = super.validate( propertyName );
+
+            if( !status.isOK() )
+            {
+                return status;
+            }
+
+            String currentPortletName = getStringProperty( PORTLET_NAME );
+            PortletDescriptorHelper helper = new PortletDescriptorHelper( getTargetProject() );
+
+            for( String portletName : helper.getAllPortletNames() )
+            {
+                if( currentPortletName.equals( portletName ) )
+                {
+                    return PortletCore.createErrorStatus( Msgs.duplicatePortletName );
+                }
+            }
+        }
+        else if( PROJECT_NAME.equals( propertyName ) )
+        {
+            final SDK sdk = SDKUtil.getSDK( getTargetProject() );
```

Comment:
This wizard has to support non-sdk projects too, so there is no need for this check.